### PR TITLE
Add const correctness and remove heap allocation

### PR DESCRIPTION
**Pull request by mark2185**
Removed pointers and added const correctness.

### CODE REVIEW

**Comment by sebojanko** [comment]
I'm interested in the benchmarks. Does this change speed things up?
I'm not approving this without the numbers.

**Review by sebojanko — APPROVED** [review]

**Comment by cizl** [comment]
eli5? Is the main goal of this PR to add some good practices with const correctness? https://stackoverflow.com/questions/136880/sell-me-on-const-correctness
Also allocating on stack should be faster, right? Any comparison?
Seems good to me, I'm approving.

**Review by cizl — APPROVED** [review]

**Comment by mark2185** [comment]
This is to drop the `new Hasher(blahblah)` part of the code. The const correctness part was because I added it somewhere, then it started breaking so I decided to fix it altogether.

I haven't yet compared performances between this and the master version.

**Comment by mark2185** [comment]
Depending on the input, the performance improves up to 10%, but it's really hit and miss.

I haven't tested with the SimpleTable, only with the Compact one, but I doubt we would have seen anything different.
